### PR TITLE
Feat/awicm3 v3.2.1 bringing OASIS with YAC

### PR DESCRIPTION
**Pull request by JanStreffing**
This branch can compile oasis with yaxt and yac on juwels; compile oifs, fesom, xios and rnfmap with oasis/yac, and provided one does manual modification of the namcouple to put the yac stack in runtime works too.

This last part should be worked on before merge.

esm_runscripts should be able to automatically create this block in a namcouple based on an interpolation method:
```
A_Precip_liquid:A_Precip_solid:A_Evap:A_Subl prec_oce:snow_oce:evap_oce:subl_oce 3 7200 2 rstas.nc EXPOUT
40320 1 126858 1 A096 feom LAG=0
P 0 P 0
LOCTRANS YAC
INSTANT
LL GC 2 YAC_CONSERV_FRACNNEI 4
CONSERV FIRST FRACAREA
NNN AVG 1
```
The first three lines are identical to old methods. `LOCTRANS YAC` does not work yet, because SCRIPR is hardcoded here: https://github.com/esm-tools/esm_tools/blob/f45e373a9aa463ce81d0b13170346e85a8df2c87/src/esm_runscripts/oasis.py#L214 

We also need to be able to place a variable number of remappings in, because YAC always does a remapping stack. It tries a method, and falls back for grid points that didn't work. The number of fallback layers is arbitrary. Here its three

`LL GC 2 YAC_CONSERV_FRACNNEI 4`
`CONSERV FIRST FRACAREA`
`NNN AVG 1`

### CODE REVIEW

**Comment by JanStreffing** [comment]
Here is some pseudocode I've stared to add here: https://github.com/esm-tools/esm_tools/blob/43afb179c0cdf441f9592e870d195b2ba104f7b5/src/esm_runscripts/oasis.py#L268

```
            elif trans.lower() in [
                "oyac_stack",
            ]:
                trafo_line += " YAC"
                #TODO: Read new var nb_stack for number of interpolations in stack
                srcgridtype = str(rgrid["oasis_grid_type"]).upper()
                #TODO: We need not only src but also dst grid type
                dstgridtype = str(rgrid["oasis_grid_type"]).upper()
                detail_line = (
                    trans.upper()
                    + srcgridtype.upper()
                    + " "
                    + dstgridtype.upper()
                    + " "
                    # + nb_stack
                    + " YAC_CONSERV "
                    # + nb_io_per_node
                )
                trafo_details += [detail_line.strip()]
                for interp in nb_stack:
                # we need logic for content of oyac stack. unless we want to make a new entry here for every possible combination of transformations
                stack_line = (
```

I think we need to define a new dict or list of remappings, have the number of remappings and we need the dst grid type. Adding additional lines after the detail line in a loop over the remappings in the stack should by easy.

**Comment by JanStreffing** [comment]
Had a chat with @mandresm. 

- In the yaml files, we can just add multiple remappings to one coupling method. 
- We add an oyac logic switch that can be read via `oyac = transformation.get("oyac", {"false"})`
- If switch is true we add line 3 of the namcouple block: 

```
LOCTRANS YAC
STOCHA
GC LL 2 YAC_CONSERV_FRACNNEI 4
CONSERV FIRST FRACAREA
NNN AVG 1
```

- yaml would look like:

```
                oyac_1stcon_nnn_stocha:
                        oyac: true
                        time_transformation: stocha
                        remapping:
                                conserv:
                                        order: "first"
                                        neighbour: fracarea
                                nnn:
                                        weighting: "avg"
                                        nb_of_neighbour: 1
```

**Comment by JanStreffing** [comment]
Draft complete, ready for review.

Branch can: 

- Download and compile YAXT and YAC,
- Compile OASIS with YAC lib (OYAC)
- Compile AWI-CM3-v3.2.1 with OYAC in all components (XIOS, RNFMAP, OIFS, FESOM2)
- Create a namcouple files with OYAC stacks based on esm_runscripts yaml files.
- Run AWI-CM3-v3.2.1 with OYAC conserv+nnn stack, optionally with stochastic coupling.

File: configs/components/oasis3mct/oasis3mct.yaml [hunk 1]
```diff
@@ -81,7 +82,23 @@ choose_version:
...
...e -j 6 install ; cd ../../ ; [ -d "yac" ] || git clone --branch=release-3.1.0 https://gitlab.dkrz.de/dkrz-sw/yac.git ; cd yac ; export YAC_ROOT=$(pwd) ; mkdir -p build ; cd build ; ../configure --enable-lib-core-only --disable-mpi-checks --with-yaxt-root=${model_dir}/lib/yaxt --with-netcdf-root=$NETCDFROOT --prefix=${model_dir}/lib/yac ; make -j 6 install ; export YAC_INCLUDE="-I${model_dir}/lib/yac/include -I${model_dir}/lib/yaxt/include"; cd ../../../util/make_dir/ ; cp make.ESMTOOLS make.inc ; make -f TopMakefileOasis3 ; cd ../../ ; mv INSTALL_OASIS.ESMTOOLS/lib/libpsmile.MPI1.a INSTALL_OASIS.ESMTOOLS/lib/libpsmile.a ; mkdir -p build/lib/psmile/scrip ; mkdir -p build/lib/psmile/mct/mpeu ; cp INSTALL_OASIS.ESMTOOLS/build-static/lib/psmile.MPI1/*.mod build/lib/psmile/. ; cp INSTALL_OASIS.ESMTOOLS/build-static/lib/scrip/*mod build/lib/psmile/scrip/. ; cp INSTALL_OASIS.ESMTOOLS/build-static/lib/mct/*mod build/lib/psmile/mct/. ; cp INSTALL_OASIS.ESMTOOLS/lib/libmct.a build/lib/psmile/mct/. ; cp INSTALL_OASIS.ESMTOOLS/lib/libmpeu.a build/lib/psmile/mct/. ; cp INSTALL_OASIS.ESMTOOLS/lib/libpsmile.a build/lib/psmile/. ; cp INSTALL_OASIS.ESMTOOLS/lib/libscrip.a build/lib/psmile/scrip/. '
```
Comment (mandresm) [inline configs/components/oasis3mct/oasis3mct.yaml]:
```suggestion
    comp_command: '
        export ESM_OASIS_DIR=${model_dir};
        cd lib;
        [ -d "yaxt" ] || rm -f yaxt-0.10.0.tar.gz && wget https://swprojects.dkrz.de/redmine/attachments/download/529/yaxt-0.10.0.tar.gz ;
        [ -d "yaxt" ] || mkdir yaxt && tar -xzvf yaxt-0.10.0.tar.gz -C yaxt --strip-components=1 ;
        rm yaxt-0.10.0.tar.gz ;
        cd yaxt; export YAXT_ROOT=$(pwd) ;
        mkdir -p build;
        cd build;
        ../configure --prefix=${model_dir}/lib/yaxt --without-regard-for-quality ;
        make -j 6 install ;
        cd ../../ ;
        [ -d "yac" ] || git clone --branch=release-3.1.0 https://gitlab.dkrz.de/dkrz-sw/yac.git ;
        cd yac ;
        export YAC_ROOT=$(pwd) ;
        mkdir -p build ;
        cd build ;
        ../configure --enable-lib-core-only --disable-mpi-checks --with-yaxt-root=${model_dir}/lib/yaxt --with-netcdf-root=$NETCDFROOT --prefix=${model_dir}/lib/yac ;
        make -j 6 install ;
        export YAC_INCLUDE="-I${model_dir}/lib/yac/include -I${model_dir}/lib/yaxt/include"; cd ../../../util/make_dir/ ;
        cp make.ESMTOOLS make.inc ;
        make -f TopMakefileOasis3 ;
        cd ../../ ; mv INSTALL_OASIS.ESMTOOLS/lib/libpsmile.MPI1.a INSTALL_OASIS.ESMTOOLS/lib/libpsmile.a ;
        mkdir -p build/lib/psmile/scrip ;
        mkdir -p build/lib/psmile/mct/mpeu ;
        cp INSTALL_OASIS.ESMTOOLS/build-static/lib/psmile.MPI1/*.mod build/lib/psmile/. ;
        cp INSTALL_OASIS.ESMTOOLS/build-static/lib/scrip/*mod build/lib/psmile/scrip/. ;
        cp INSTALL_OASIS.ESMTOOLS/build-static/lib/mct/*mod build/lib/psmile/mct/. ;
        cp INSTALL_OASIS.ESMTOOLS/lib/libmct.a build/lib/psmile/mct/. ;
        cp INSTALL_OASIS.ESMTOOLS/lib/libmpeu.a build/lib/psmile/mct/. ;
        cp INSTALL_OASIS.ESMTOOLS/lib/libpsmile.a build/lib/psmile/. ;
        cp INSTALL_OASIS.ESMTOOLS/lib/libscrip.a build/lib/psmile/scrip/. '
```
Feel free to reject this suggestion, just to improve readibility

File: configs/components/oasis3mct/oasis3mct.yaml [hunk 2]
```diff
@@ -81,7 +82,23 @@ choose_version:
...
...nstall" ; cd ../../ ; [ -d "yac" ] || git clone --branch=release-3.1.0 https://gitlab.dkrz.de/dkrz-sw/yac.git ; cd yac ; export YAC_ROOT=$(pwd) ; mkdir -p build ; cd build ; echo "../configure --enable-lib-core-only --disable-mpi-checks --with-yaxt-root=${model_dir}/lib/yaxt --with-netcdf-root=$NETCDFROOT --prefix=${model_dir}/lib/yac ; make -j 6 install" ; export YAC_INCLUDE="-I${model_dir}/lib/yac/include -I${model_dir}/lib/yaxt/include"; cd ../../../util/make_dir/ ; cp make.ESMTOOLS make.inc ; make -f TopMakefileOasis3 ; cd ../../ ; mv INSTALL_OASIS.ESMTOOLS/lib/libpsmile.MPI1.a INSTALL_OASIS.ESMTOOLS/lib/libpsmile.a ; mkdir -p build/lib/psmile/scrip ; mkdir -p build/lib/psmile/mct/mpeu ; cp INSTALL_OASIS.ESMTOOLS/build-static/lib/psmile.MPI1/*.mod build/lib/psmile/. ; cp INSTALL_OASIS.ESMTOOLS/build-static/lib/scrip/*mod build/lib/psmile/scrip/. ; cp INSTALL_OASIS.ESMTOOLS/build-static/lib/mct/*mod build/lib/psmile/mct/. ; cp INSTALL_OASIS.ESMTOOLS/lib/libmct.a build/lib/psmile/mct/. ; cp INSTALL_OASIS.ESMTOOLS/lib/libmpeu.a build/lib/psmile/mct/. ; cp INSTALL_OASIS.ESMTOOLS/lib/libpsmile.a build/lib/psmile/. ; cp INSTALL_OASIS.ESMTOOLS/lib/libscrip.a build/lib/psmile/scrip/. '
```
Comment (mandresm) [inline configs/components/oasis3mct/oasis3mct.yaml:93]:
```suggestion
```
Do you still need this comment? Otherwise you can remove it.

File: configs/components/oasis3mct/oasis3mct.yaml [hunk 3]
```diff
@@ -81,7 +82,23 @@ choose_version:
...
+    install_libs:
+    - INSTALL_OASIS.ESMTOOLS/lib/libpsmile.a
+    - INSTALL_OASIS.ESMTOOLS/lib/libmct.a
+    - INSTALL_OASIS.ESMTOOLS/lib/libmpeu.a
+    - INSTALL_OASIS.ESMTOOLS/lib/libscrip.a
+    #clean_command: 'rm -rf include lib/libpsmile.a lib/libscrip.a lib/libmct.a lib/libmpeu.a'
```
Comment (mandresm) [inline configs/components/oasis3mct/oasis3mct.yaml:99]:
```suggestion
```
Is this comment needed still?

File: configs/machines/juwels.yaml [hunk 4]
```diff
@@ -12,7 +12,7 @@ accounting: true
 #compiler_mpi: intel_2019a_psmpi2019
 #iolibraries: system_libs
 # for FOCI/FOCIOIFS use
-compiler_mpi: intel2019_impi2019
+compiler_mpi: intel2022_ompi2022
```
Comment (mandresm) [inline configs/machines/juwels.yaml:15]: Why is this needed here when it's already defined in `awicm3.yaml`? Is it because one of the bugs we were talking about right?

**Review by mandresm — CHANGES_REQUESTED** [review]
Hey Jan!

Great work, only some comments. I've done some black formatting to oasis.py and merged release into the branch solving a few conflicts in oifs.yaml, but nothing serious.

File: configs/components/oifs/oifs.env.yaml [hunk 5]
```diff
@@ -321,8 +321,8 @@ compiletime_environment_changes:
         LAPACK_LIB_DEFAULT: '"-L$MKLROOT/lib/intel64 -lmkl_intel_lp64 -lmkl_core -lmkl_sequential"'
         # oasis
         OIFS_OASIS_BASE: '$(pwd)/oasis'
-        OIFS_OASIS_INCLUDE: '"-I$OIFS_OASIS_BASE/build/lib/psmile -I$OIFS_OASIS_BASE/build/lib/psmile/scrip -I$OIFS_OASIS_BASE/build/lib/psmile/mct -I$OIFS_OASIS_BASE/build/lib/psmile/mct/mpeu"'
-        OIFS_OASIS_LIB: '"-L$OIFS_OASIS_BASE/build/lib/psmile -L$OIFS_OASIS_BASE/build/lib/psmile/scrip -L$OIFS_OASIS_BASE/build/lib/psmile/mct -L$OIFS_OASIS_BASE/build/lib/psmile/mct/mpeu -lpsmile -lmct -lmpeu -lscrip"'
+        OIFS_OASIS_INCLUDE: '"-I$OIFS_OASIS_BASE/build/lib/psmile -I$OIFS_OASIS_BASE/build/lib/psmile/scrip -I$OIFS_OASIS_BASE/build/lib/psmile/mct -I$OIFS_OASIS_BASE/build/lib/psmile/mct/mpeu -I$OIFS_OASIS_BASE/lib/include/yac -I$OIFS_OASIS_BASE/lib/yaxt/include"'
+        OIFS_OASIS_LIB: '"-L$OIFS_OASIS_BASE/build/lib/psmile -L$OIFS_OASIS_BASE/build/lib/psmile/scrip -L$OIFS_OASIS_BASE/build/lib/psmile/mct -L$OIFS_OASIS_BASE/build/lib/psmile/mct/mpeu -L$OIFS_OASIS_BASE/lib/yac/lib -L$OIFS_OASIS_BASE/lib/yaxt/lib -lpsmile -lmct -lmpeu -lscrip -lyac_utils -lyac_core -lyaxt -lyaxt_c -qmkl"'
```
Comment (mandresm) [inline configs/components/oifs/oifs.env.yaml]: Are this changes okay in terms of back-compatibility?

It's clearly having an effect in esm_tests (as it would be expected): https://github.com/esm-tools/esm_tools/actions/runs/8205605052/job/22442824281?pr=1127#step:8:65

**Comment by mandresm** [comment]
This lines in the tests are worrying: https://github.com/esm-tools/esm_tools/actions/runs/8205605052/job/22442824281?pr=1127#step:8:103

Are you intentionally killing the `-DOIFS_COUPLED=ON -DFESOM_COUPLED=ON` in the cmake command?

File: configs/components/oasis3mct/oasis3mct.yaml (hunk 3 above)
Comment (JanStreffing) [reply configs/components/oasis3mct/oasis3mct.yaml:99]: It can be helpful. Rebuilding only the old parts of OASIS takes about 1min. Rebuilding YAXT+YAC+OASIS takes about 15. If one does developments in OASIS (like I do for stochastic coupling), this different clean command is good to have.

File: configs/components/oasis3mct/oasis3mct.yaml (hunk 2 above)
Comment (JanStreffing) [reply configs/components/oasis3mct/oasis3mct.yaml:93]: Goes with the different clean command from below.

File: configs/machines/juwels.yaml (hunk 4 above)
Comment (JanStreffing) [reply configs/machines/juwels.yaml:15]: I don't recall at the moment... One of us will have to try this out. Want to flip a coin who? ;-)

File: configs/components/oifs/oifs.env.yaml (hunk 5 above)
Comment (JanStreffing) [reply configs/components/oifs/oifs.env.yaml]: No, I think you found the part that was breaking the awicm3-v3.1.1 compile on this branch. I will need to develop a different solution!

**Comment by JanStreffing** [comment]
> This lines in the tests are worrying: https://github.com/esm-tools/esm_tools/actions/runs/8205605052/job/22442824281?pr=1127#step:8:103
> 
> Are you intentionally killing the `-DOIFS_COUPLED=ON -DFESOM_COUPLED=ON` in the cmake command?

This is also strange indeed. That was not supposed to happen. FESOM2 still needs to be compiled with these, as they include the OASIS interface ifdef preprocessor blocks in the model.

File: configs/components/oifs/oifs.env.yaml (hunk 5 above)
Comment (joakimkjellsson) [reply configs/components/oifs/oifs.env.yaml]: Won't this cause a compile error if I try to compile OpenIFS on Juwels without YAC, i.e. OASIS3-MCT4 or something?  
Could we instead make a variable `OIFS_OASIS_YAC_LIB = -lyac_utils -lyac_core -lyaxt -lyaxt_c` which we can add to `OIFS_OASIS_LIB` above? 
Then we can do something like this in `oasis.yaml`:

```python
choose_oasis.with_yac:
    1: 
        add_export_vars:
            OIFS_OASIS_YAC_LIB: "-lyac_utils -lyac_core -lyaxt -lyaxt_c"
    0:
        add_export_vars:
            OIFS_OASIS_YAC_LIB: ""
```

Comment (JanStreffing) [reply configs/components/oifs/oifs.env.yaml]: That's a good suggestion. I will try this.

Comment (JanStreffing) [reply configs/components/oifs/oifs.env.yaml]: In the oasis.3vn.yaml i added:
```
add_compiletime_environment_changes:
        choose_oasis3mct.with_yac:
                True:
                        OIFS_OASIS_YAC_LIBS: "-lyac_utils -lyac_core -lyaxt -lyaxt_c"
                        OIFS_OASIS_YAC_INC_PATH: "-I$OIFS_OASIS_BASE/lib/include/yac -I$OIFS_OASIS_BASE/lib/yaxt/include"
                        OIFS_OASIS_YAC_LIB_PATH: "-L$OIFS_OASIS_BASE/lib/yac/lib -L$OIFS_OASIS_BASE/lib/yaxt/lib"
                False:
                        OIFS_OASIS_YAC_LIBS: ""
                        OIFS_OASIS_YAC_INC_PATH: ""
                        OIFS_OASIS_YAC_LIB_PATH: ""
```
I get Key oasis3mct.with_yac was not defined. Same with `choose_with_yac`.

File: configs/machines/juwels.yaml (hunk 4 above)
Comment (JanStreffing) [reply configs/machines/juwels.yaml:15]: I recall now. I did this because when you load 2019 you get a message that it will be deleted soon:
  **This stage is deprecated and will be removed soon. Please consider moving to a new stage (2024 or newer)**

File: configs/components/oifs/oifs.env.yaml (hunk 5 above)
Comment (JanStreffing) [reply configs/components/oifs/oifs.env.yaml]: Any advice on what I'm doing wrong here?

Comment (joakimkjellsson) [reply configs/components/oifs/oifs.env.yaml]: Did you do something like this in `oasis3mct.yaml`:
```python
with_yac: False
...
choose_version:
   ...
   - '6.0-YAC':
       with_yac: True
```
or how is `with_yac` defined?

Comment (JanStreffing) [reply configs/components/oifs/oifs.env.yaml]: Yes:

```
  6.0-YAC:
    # this version is taken directly from the CERFACS gitlab
    # it compiles in the standard way, i.e. not with cmake
    with_yac: True
```

Comment (JanStreffing) [reply configs/components/oifs/oifs.env.yaml]: When people are back from holiday, I would like to have some help on this. I've been doing ugly workarounds for now. @mandresm @pgierz

Comment (mandresm) [reply configs/components/oifs/oifs.env.yaml]: should be sorted now after the last commit

**Comment by JanStreffing** [comment]
I tired hard making the new method the default for all remappings in AWI-CM3, but I still have not managed for the runoff mapper. In the meantime I would like to finish this one, even if the feature is not turned on by default here.

**Comment by JanStreffing** [comment]
A merge conflict recently seems to have come in for oasis.py
@pgierz, could you ensure that the logger changes made in release since I branched off here, make it into this branch?
Then the conflict should be resolved.

**Comment by JanStreffing** [comment]
I added a runscript that uses 100% yac remapping for AWI-CM3. By default, this is still turned off.
It will be checked with the next major version where I will also make experiments. For the minor version, only tech changes.

**Comment by JanStreffing** [comment]
I will double check one more time that none of the OASIS/SCRIP setups are negativly affected.

**Comment by JanStreffing** [comment]
Found some minor hickups on juwels still. Not ready yet.

**Comment by joakimkjellsson** [comment]
Thanks for your work on this @JanStreffing. If you're convinced it works for AWI-CM3, then I'd be curious to try for FOCI as well.

**Comment by JanStreffing** [comment]
compiles and run v3.2.1 and v3.1 on levante. On Juwels, compiles, but OpenIFS crashes with:
`160:  ABORT!   17 SUMSC: WORD KIND LENGTH PROBLEM`
This will have to do with me switching from 2019 to 2022 libs on juwels, I think. Still more work to be done..

**Comment by JanStreffing** [comment]
On levante:
```
 --- Set up machine-specific constants----------------------------------
 NINTLEN=  4 NREALEN=  8 NLOGLEN=  8 NDBLLEN= 16
```

On juwels:
```
 --- Set up machine-specific constants----------------------------------
 NINTLEN=  0 NREALEN=  0 NLOGLEN=  8 NDBLLEN= 16
 ABOR1 CALLED
 SUMSC: WORD KIND LENGTH PROBLEM
```

**Comment by JanStreffing** [comment]
It is done! This branch now runs AWI-CM3 v3.2 and v3.2.1 on juwels and levante. With the former, one can use SCRIP rmp files, with the latter one can use SCRIP or YAC rmp files, provided all fields have corner and correct names. e.g. `/work/ab0246/a270092/input/oasis/cy43r3/TCO95-CORE2/grids.nc`. Here lies a quick and dirty script to modify old grids.nc files for the new standard: `/work/ab0246/a270092/postprocessing/oasis_yac/adopt_rnf_from_scrip_to_yac.py`. Files of the new standard are backwards compatible with existing rmp files. I will add this to https://github.com/JanStreffing/ocp-tool sonn TM

@mandresm Ready from my side.

**Comment by JanStreffing** [comment]
Would it be possible to have this merged by Monday? I would like to quickly throw out v3.2.2 with some minore fixes next week before I start working on v3.3 again.

File: configs/components/oasis3mct/oasis3mct.env.yaml [hunk 6]
```diff
@@ -4,9 +4,22 @@ add_compiletime_environment_changes:
...
+                                FCFLAGS: '"-fPIC"'
                 levante:
                         add_export_vars:
-                                OASIS_FFLAGS: '"-march=core-avx2 -mtune=core-avx2"'
+                                OASIS_FFLAGS: '"-fPIC"'
+                                OASIS_CFLAGS: '"-fPIC"'
+                                CFLAGS: '"-fPIC"'
+                                CCFLAGS: '"-fPIC"'
+                                FFLAGS: '"-fPIC"'
+                                FCFLAGS: '"-fPIC"'
```
Comment (mandresm) [inline configs/components/oasis3mct/oasis3mct.env.yaml:22]: This is introducing changes in all models using Oasis, what does this change exactly do?

File: configs/components/oasis3mct/oasis3mct.env.yaml [hunk 7]
```diff
@@ -16,3 +29,19 @@ add_compiletime_environment_changes:
...
+                yac_lib_path: "-L$OIFS_OASIS_BASE/lib/yac/lib -L$OIFS_OASIS_BASE/lib/yaxt/lib"
+        False:
+                yac_libs: ""
+                yac_inc_path: ""
+                yac_lib_path: ""
+
+
+add_runtime_environment_changes:
+        add_export_vars:
+                - 'LD_LIBRARY_PATH=${model_dir}/oasis/lib/yaxt/lib/:${model_dir}/oasis/lib/yac/lib/:$LD_LIBRARY_PATH'
```
Comment (mandresm) [inline configs/components/oasis3mct/oasis3mct.env.yaml]: Shouldn't this go into the choose block above?

File: configs/setups/awicm3/awicm3.yaml [hunk 8]
```diff
@@ -368,8 +387,13 @@ fesom:
...
                         nproc: 72
-
-        comp_command: mkdir -p build; cd build; cmake -DOIFS_COUPLED=ON -DFESOM_COUPLED=ON ..;   make install -j `nproc --all`
+        choose_general.oasis_with_yac:
+            True:
+                comp_command: mkdir -p build; cd build; cmake -DOIFS_COUPLED=ON -DFESOM_COUPLED=ON -DOASIS_WITH_YAC=ON -DCMAKE_INSTALL_PREFIX=../  ..;   make install -j `nproc --all`
+            False:
+                comp_command: mkdir -p build; cd build; cmake -DOIFS_COUPLED=ON -DFESOM_COUPLED=ON -DCMAKE_INSTALL_PREFIX=../ ..;   make install -j `nproc --all`
+            "*":
+                comp_command: echo "please define oasis_with_yac boolean in awicm3.yaml"
```
Comment (mandresm) [inline configs/setups/awicm3/awicm3.yaml]:
```suggestion
                comp_command: "missing"
                error:
                    "Missing variable":
                        message: "Please define ``oasis_with_yac`` boolean in ``awicm3.yaml``"
                        ask_user_to_continue: False
```

**Review by mandresm — CHANGES_REQUESTED** [review]
Hi @JanStreffing, only a few questions about changes happening in the tests. I can take care of rewritting anything it's needed to be included in a choose block after I get the answers.

File: configs/components/oasis3mct/oasis3mct.env.yaml (hunk 6 above)
Comment (JanStreffing) [reply configs/components/oasis3mct/oasis3mct.env.yaml:22]: Generate position-independent code (PIC) suitable for use in a shared library, if supported for the target machine. Such code accesses all constant addresses through a global offset table (GOT). ([link](https://gcc.gnu.org/onlinedocs/gcc/Code-Gen-Options.html#Code-Gen-Options))

This is needed for the new FESOM2 versions after production_DE, as well as for oifs 48r1 and should not result in any other changes. It just makes cmake happy to link the oasis libs.

File: configs/components/oasis3mct/oasis3mct.env.yaml (hunk 7 above)
Comment (JanStreffing) [reply configs/components/oasis3mct/oasis3mct.env.yaml]: It does no harm here, but it makes more sense up there. I will try if that works.

File: configs/components/oasis3mct/oasis3mct.env.yaml (hunk 6 above)
Comment (mandresm) [reply configs/components/oasis3mct/oasis3mct.env.yaml:22]: ah, I see, and it is supported in Levante and Juwels, and that's why we are seeing this here right? Thanks for the info :)

**Comment by JanStreffing** [comment]
Hey @mandresm, anything else that needs to be done for this? Are you happy with the fPIC flag as is?

**Comment by mandresm** [comment]
Hi @Jan, fPIC we can leave it as it is. The reason why I did not merge, bump or approve was because I was having problems with the add_runtime_environment_changes. Turns out that the problem was not there, but it was with the fact that awicm3.yaml was not defining the version of oasis, and consequently the version used was 2.5, and `with_yac` was False. Now that's fixed and all is working as it should and I can see the LD_LIBRARY_PATH for yaxt and yac only on awicm3-v3.2.1.

Now I'm going through the merging workflow. Should be done in an hour top

**Comment by jan** [comment]
@mandresm keep up the good work!

**Comment by mandresm** [comment]
#approve-changes

**Comment by mandresm** [comment]
> @mandresm keep up the good work!

Thanks and sorry for the wrong tagging!

**Comment by mandresm** [comment]
#bump #minor

**Review by mandresm — APPROVED** [review]
